### PR TITLE
Rel 813210 revert old stuck changes v2

### PR DESCRIPTION
**Pull request by kacperniwczykr1**

### CODE REVIEW

File: lib/MD5HashStream.cs [hunk 1]
```diff
@@ -233,7 +228,7 @@ public void CalculateMd5(MemoryManager memoryManager, Action checkCancellation)
         /// <c>TResult</c> parameter contains the total number of bytes read into the buffer.</returns>
         public async Task<int> ReadAsync(long readOffset, byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
-            await this.WaitOnSemaphoreAsync(cancellationToken).ConfigureAwait(false);
```
Comment (KamilKalitaRel) [inline lib/MD5HashStream.cs:236]: I'd leave `.ConfigureAwait(false)` (in this and other places).

File: test/DMLibTest/Cases/AllTransferDirectionTest.cs [hunk 2]
```diff
@@ -127,7 +127,22 @@ private static void PrepareSourceData()
                 }
             }
 
-            DMLibTestContext.DestType = DMLibDataType.AppendBlob;
```
Comment (KamilKalitaRel) [inline test/DMLibTest/Cases/AllTransferDirectionTest.cs:130]: I don't remember this changes... Was it related to some failing tests?

Comment (kacperniwczykr1) [reply test/DMLibTest/Cases/AllTransferDirectionTest.cs:130]: I would need to ask someone to help me with local test of DMLib to confirm if this doesn't affect any tests. For now I'm getting error on almost every test

**Review by KlaudiuszBryjaRelativity — DISMISSED** [review]

**Review by KamilKalitaRel — DISMISSED** [review]

**Review by KlaudiuszBryjaRelativity — CHANGES_REQUESTED** [review]

File: test/DMLibTest/Cases/AllTransferDirectionTest.cs (hunk 2 above)
Comment (kacperniwczykr1) [reply test/DMLibTest/Cases/AllTransferDirectionTest.cs:130]: I've reverted it when I was working on my failing tests, I will revert it to old state again and re run this test case

Comment (KlaudiuszBryjaRelativity) [reply test/DMLibTest/Cases/AllTransferDirectionTest.cs:130]: This line is removed. Test passed without this change.

**Review by KlaudiuszBryjaRelativity — APPROVED** [review]

**Review by KamilKalitaRel — DISMISSED** [review]

**Review by KamilKalitaRel — APPROVED** [review]

**Review by KlaudiuszBryjaRelativity — APPROVED** [review]

**Pull request merged**